### PR TITLE
Updates to use classes

### PR DESCRIPTION
**Pull request by vizhur**
update to classes

### CODE REVIEW

File: scripts/docs/generate_asset_documentation.py [hunk 1]
```diff
@@ -10,184 +10,305 @@
...
+        doc.add_heading("Docker build context", level=2)
+        context = {}
+        dockerfile = ""
+        for context_file in search(str(self._asset_config.extra_config_as_object().context_dir_with_path) + "/*"):
+            content = ""
+            with open(context_file, "r") as f:
+                # do not expect nested structure for now
+                content = f.read()
+            filename = os.path.basename(context_file)
+            if filename.lower() == "dockerfile":
```
Comment (elesel) [inline scripts/docs/generate_asset_documentation.py:147]: Please consider using _asset_config.extra_config_as_object()'s (EnvironmentConfig's) dockerfile properties.

File: scripts/docs/generate_asset_documentation.py [hunk 2]
```diff
@@ -10,184 +10,305 @@
...
+        return Path(f"{self.type}s")
+
+    @property
+    def fullpath(self):
+        return f"{self.directory}/{self.filename}.md"
+
+    @property
+    def doc(self):
+        _doc = snakemd.new_doc()
+        return _doc
```
[the remark this answers is not in the document]
Comment (elesel) [reply scripts/docs/generate_asset_documentation.py:76]: So it's just there to return an empty doc?

File: scripts/docs/generate_asset_documentation.py [hunk 3]
```diff
@@ -10,183 +10,306 @@
...
 
+supported_asset_types=["environment", "component", "model"]
+default_category = "Uncategorized"
+
+class AssetInfo:
+    _asset_config = None
+    _asset = None
+    _doc = None
+
+    def __init__(self, asset_config):
```
[the remark this answers is not in the document]
Comment (vizhur) [reply scripts/docs/generate_asset_documentation.py:24]: sorry, I'm py35 guy :)

File: scripts/docs/generate_asset_documentation.py [hunk 4]
```diff
@@ -10,184 +10,305 @@
 from pathlib import Path
 from glob import glob as search
 
+supported_asset_types = ["environment", "component", "model"]
+default_category = "Uncategorized"
```
[the remark this answers is not in the document]
Comment (vizhur) [reply scripts/docs/generate_asset_documentation.py:14]: I want to switch to AssetType enum, had troubles going between strings and enumerations so string-ified everything for now

File: scripts/docs/generate_asset_documentation.py [hunk 5]
```diff
@@ -10,184 +10,305 @@
 from pathlib import Path
 from glob import glob as search
 
+supported_asset_types = ["environment", "component", "model"]
```
[the remark this answers is not in the document]
Comment (vizhur) [reply scripts/docs/generate_asset_documentation.py:13]: I want to switch to AssetType enum, had troubles going between strings and enumerations so string-ified everything for now

File: scripts/docs/generate_asset_documentation.py [hunk 6]
```diff
@@ -10,184 +10,305 @@
...
+    def __init__(self, asset_config):
+        self._asset_config = asset_config
+        yaml = YAML()
+        with open(self._asset_config.spec_with_path, 'r') as f:
+            self._asset = yaml.load(f)
+        self._doc = snakemd.new_doc()
+
+    @property
+    def type(self):
+        return self._asset_config.type.value
```
[the remark this answers is not in the document]
Comment (vizhur) [reply scripts/docs/generate_asset_documentation.py:35]: we only need a string value here, I'll switch to the type thing in a following PR(s)

File: scripts/docs/generate_asset_documentation.py [hunk 7]
```diff
@@ -10,184 +10,305 @@
...
+    def directory(self):
+        return Path(f"{self.type}s")
+
+    @property
+    def fullpath(self):
+        return f"{self.directory}/{self.filename}.md"
+
+    @property
+    def doc(self):
+        _doc = snakemd.new_doc()
```
[the remark this answers is not in the document]
Comment (vizhur) [reply scripts/docs/generate_asset_documentation.py:75]: not really, I can throw NotImplemented instead?

File: scripts/docs/generate_asset_documentation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vizhur) [reply scripts/docs/generate_asset_documentation.py:76]: either empty doc to not fail a script, all child classes should implement it though

File: scripts/docs/generate_asset_documentation.py (hunk 1 above)
Comment (vizhur) [reply scripts/docs/generate_asset_documentation.py:148]: that's in my todo from your comment to a previous PR :)

File: scripts/docs/generate_asset_documentation.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (elesel) [reply scripts/docs/generate_asset_documentation.py:78]: Yeah, that would be great!

**Review by elesel — APPROVED** [review]

**Review by elesel — APPROVED** [review]

**Pull request merged**